rethinkdb/datadog_checks/rethinkdb/document_db/query.py

```diff
@@ -0,0 +1,118 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from typing import Any, Callable, Iterator, List, Mapping, Sequence, Tuple
+
+from datadog_checks.base import AgentCheck
+from datadog_checks.base.log import CheckLoggingAdapter
+
+from .types import Enumeration, Group, Metric, MetricSpec, Modifier
+from .utils import dotted_join, lookup_dotted, to_time_elapsed
+
+
+class DocumentQuery(object):
+    """
+    A helper for retrieving metrics from document-oriented ("JSON") databases.
+    """
+
+    def __init__(
+        self,
+        source,  # type: Callable[..., Iterator[Tuple[Any, List[str]]]]
+        name,  # type: str
+        prefix,  # type: str
+        metrics=None,  # type: List[MetricSpec]
+        enumerations=None,  # type: List[Enumeration]
+        groups=None,  # type: List[Group]
+    ):
+        self.source = source
+        self.name = name
+        self.prefix = prefix
+        self.metrics = [] if metrics is None else metrics
+        self.enumerations = [] if enumerations is None else enumerations
+        self.groups = [] if groups is None else groups
+
+    def _make_metric_from_spec(self, document, spec, tags, logger):
+        # type: (Any, MetricSpec, List[str], CheckLoggingAdapter) -> Metric
+        logger.trace('make_metric_from_spec %r', spec)
+
+        path = spec['path']
+        name = spec.get('name', path)
+        value = lookup_dotted(document, path=path)
+
+        if 'modifier' in spec and spec['modifier'] is not None:
+            value = self._modify(value, modifier=spec['modifier'], logger=logger)
+
+        if not isinstance(value, (int, float)):  # pragma: no cover
+            raise RuntimeError('Expected float or int, got {!r} of type {}', value, type(value))
+
+        name = dotted_join((self.prefix, name))
+
+        return {'type': spec['type'], 'name': name, 'value': value, 'tags': tags}
+
+    def _make_metrics_from_enumeration(self, document, enumeration, tags, logger):
+        # type: (Any, Enumeration, List[str], CheckLoggingAdapter) -> Iterator[Metric]
+        logger.trace('make_metrics_from_enumeration enumeration=%r', enumeration)
```

Comment:
If those logs can help support cases, I think `debug` is more appropriate.

No integration use `trace` level for now and IMHO it's better to stay consistent with other integrations and use `debug` instead.

Reply:
Hmm, I actually wonder whether these logs are useful. They're too noisy to be debug logs (we already log the entire document before making metrics), so I'm thinking we could just drop these.